blocks/bricks/cost.py

```diff
@@ -18,6 +26,11 @@ def apply(self, y, y_hat):
         return self.cost_matrix.application_method(
             self, y, y_hat).sum(axis=1).mean()
 
+    @abstractmethod
+    @application
```

Comment:
Should do, I've used it before. It's just he order that becomes important
usually, but in this case they're complementary, so it should be good.
On Jan 21, 2015 4:36 AM, "Dmitry Bogdanov" <notifications@github.com> wrote:

> In blocks/bricks/cost.py
> <https://github.com/bartvm/blocks/pull/124#discussion_r23287283>:
>
> > @@ -18,6 +26,11 @@ def apply(self, y, y_hat):
> >          return self.cost_matrix.application_method(
> >              self, y, y_hat).sum(axis=1).mean()
> >
> > +    @abstractmethod
> > +    @application
>
> Did you try that such nested decorators work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bartvm/blocks/pull/124/files#r23287283>.
>